comptests/motr/test_motr_di.py

```diff
@@ -0,0 +1,126 @@
+    Part before yield will be invoked prior to each test case and
+    part after yield will be invoked after test call i.e as teardown.
+    """
+    request.cls.log = logging.getLogger(__name__)
+    request.cls.log.info("STARTED: Setup test operations.")
+    request.cls.secure_range = secrets.SystemRandom()
+    request.cls.nodes = CMN_CFG["nodes"]
+    request.cls.M0CRATE_WORKLOAD_YML = os.path.join(
+        os.getcwd(), "config/motr/sample_m0crate.yaml")
+    request.cls.M0CRATE_TEST_CSV = os.path.join(
+        os.getcwd(), "config/motr/m0crate_tests.csv")
+    with open(request.cls.M0CRATE_TEST_CSV) as CSV_FH:
+        request.cls.CSV_DATA = [row for row in csv.DictReader(CSV_FH)]
+    request.cls.log.info("ENDED: Setup test suite operations.")
+    yield
+    request.cls.log.info("STARTED: Test suite Teardown operations")
+    request.cls.log.info("ENDED: Test suite Teardown operations")
+
+
+class TestCorruptDataDetection:
+    """Test suite aimed at verifying detection of data corruption in degraded mode.
+    Detection supported for following entities in Normal and degraded mode.
+    1. Checksum
+    2. Data blocks
+    3. Parity
+    """
+
+    @classmethod
+    def setup_class(cls):
+        """ Setup class for running Motr tests"""
+        logger.info("STARTED: Setup Operation")
+        cls.motr_obj = MotrCoreK8s()
+        cls.m0kv_cfg = config_utils.read_yaml("config/motr/m0kv_test.yaml")
+        logger.info("ENDED: Setup Operation")
+
+    def teardown_class(self):
+        """Teardown Node object"""
+        for con in self.connections:
+            con.disconnect()
+        del self.motr_obj
+
+    def m0cp_corrupt_m0cat(self):
+        logger.info("STARTED: Verify multiple m0cp/m0cat operation")
+        infile = TEMP_PATH + 'input'
+        outfile = TEMP_PATH + 'output'
+        node_pod_dict = self.motr_obj.get_node_pod_dict()
+        motr_client_num = self.motr_obj.get_number_of_motr_clients()
+        for client_num in range(motr_client_num):
+            for node in node_pod_dict:
+                count_list = ['1', '2', '4', '4', '4', '2', '4', '4', '250',
+                              '2', '4', '2', '3', '4', '8', '4', '1024']
+                bsize_list = ['4K', '4K', '4K', '8K', '16K', '64K', '64K', '128K',
+                              '4K', '1M', '1M', '4M', '4M', '4M', '4M', '16M', '1M']
+                layout_ids = ['1', '1', '1', '2', '3', '5', '5', '6', '1',
+                              '9', '9', '11', '11', '11', '11', '13', '9']
+                for b_size, count, layout in zip(bsize_list, count_list, layout_ids):
+                    object_id = str(self.system_random.randint(1, 100)) + ":" + \
+                                str(self.system_random.randint(1, 100))
+                    self.motr_obj.dd_cmd(b_size, count, infile, node)
+                    self.motr_obj.cp_cmd(b_size, count, object_id, layout, infile, node, client_num)
```

Comment:
How are we corrupting the data here using cp_cmd? Or should we use cp_corrupt_cmd?

Reply:
@zohebkhann  Per Shankar right now he is going to add a switch and hook the datablock corruption code in CP. For long term I want DC command to be created for this purpose specifically.